src/components/popover/popover.tsx

```diff
@@ -19,13 +26,30 @@ export class Popover {
     @Prop()
     public visible = false;
 
+    /**
+     * Emits an event when the component is closing
+     */
+    @Event()
+    private close: EventEmitter<void>;
+
     @Element()
     private host: HTMLLimelPopoverElement;
 
     private portalId: string;
 
     constructor() {
         this.portalId = createRandomString();
+        this.globalClickListener = this.globalClickListener.bind(this);
+    }
+
+    public componentDidLoad() {
```

Comment:
This is only called once, but `disconnectedCallback` can be called multiple times, so we should probably use `connectedCallback` instead :thinking:

Reply:
The problem with `connectedCallback` is that it's called too early in the current implementation. Thus a close event is emitted before it has rendered.

But you had some other really good comments that might resolve this so I will look in to it.

Ideally I would prefer to use `disconnectedCallback` :)

Reply:
Aha! Yeah I guess it maybe will emit when you try to open it? :smile:

Reply:
Or maybe not since I now saw that we checked for `visible` as well before emitting :thinking:

Reply:
Unless you send in visible as a prop to the component 😛

Reply:
We could also setup the listener when the popover opens, and then remove the listener when it has been closed :slightly_smiling_face: That way we won't have to listen for events and have code running when there is no need for it